src/config/enrichment_table.rs

```diff
@@ -1,21 +1,79 @@
 use enum_dispatch::enum_dispatch;
+use serde::Serialize;
 use vector_lib::config::GlobalOptions;
-use vector_lib::configurable::{configurable_component, NamedComponent};
+use vector_lib::configurable::{configurable_component, Configurable, NamedComponent, ToValue};
+use vector_lib::id::Inputs;
 
 use crate::enrichment_tables::EnrichmentTables;
 
+use super::dot_graph::GraphConfig;
+use super::{SinkConfig, SinkOuter};
+
 /// Fully resolved enrichment table component.
 #[configurable_component]
 #[derive(Clone, Debug)]
-pub struct EnrichmentTableOuter {
+pub struct EnrichmentTableOuter<T>
+where
+    T: Configurable + Serialize + 'static + ToValue + Clone,
+{
     #[serde(flatten)]
     pub inner: EnrichmentTables,
+    #[configurable(derived)]
+    #[serde(default, skip_serializing_if = "vector_lib::serde::is_default")]
+    pub graph: GraphConfig,
+    #[configurable(derived)]
+    #[serde(
+        default = "Inputs::<T>::default",
+        skip_serializing_if = "Inputs::is_empty"
+    )]
+    pub inputs: Inputs<T>,
 }
 
-impl EnrichmentTableOuter {
-    pub fn new<I: Into<EnrichmentTables>>(inner: I) -> Self {
+impl<T> EnrichmentTableOuter<T>
+where
+    T: Configurable + Serialize + 'static + ToValue + Clone,
+{
+    pub fn new<I, IET>(inputs: I, inner: IET) -> Self
+    where
+        I: IntoIterator<Item = T>,
+        IET: Into<EnrichmentTables>,
+    {
         Self {
             inner: inner.into(),
+            graph: Default::default(),
+            inputs: Inputs::from_iter(inputs),
+        }
+    }
+
+    pub fn as_sink(&self) -> Option<SinkOuter<T>> {
```

Comment:
I think this is a reasonable way to do it the way things are currently designed, but it definitely shows the limitations of how we do things with `SinkOuter`, etc.

If I were to re-approach it, I think I would limit `SinkOuter` to config deserialization (and drop the generic to just `String`), and map to some more granular things to build the topology (e.g. something for inputs, something for healthchecks, etc). That way we could try to unify where we handle "things that have inputs", etc.

You certainly don't need to do any of that, but it could be something to explore if you'd like to make this feel cleaner.

Reply:
Yeah, not sure how often components such as this one would come up, but it definitely makes sense for components such as this one, that don't really fit just one type.

I can think about it, but I think that would be a too big undertaking for this PR.